### DISCUSSION TITLE
Welcome to flagsmith-api Discussions!

### DISCUSSION THREAD

**Discussion by dabeeeenster** [Flagsmith/flagsmith#69] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by khirod-dev** [comment]
I am recently started to look into Bullet Train . i had few questions 
1> when we change any feature value in Bullet Train will that auto pushed to UI ? or only on UI refresh the new value will be fetched via API ?
2> when we define a segment ( let's say i have that defined based on user businessUint, Employee-Id) , in Bullet Train . Now when user logs in to My application ,How do I send those values to BulletTrain that I can get the feature value for that user details.

**Comment by dabeeeenster** [comment]
Hi @khirodAsurion 

1. You would need to refresh your UI / make another call to getFlags() to retrieve the new flag values. 
2. You define Traits against an Identity (https://docs.flagsmith.com/managing-identities/#identity-traits)

**Reply by khirod-dev** [reply]
Thanks @dabeeeenster  .That helped . While I was trying to identify a user and set a trait i see fowling error :  

```**Attempted to setTrait a user before calling flagsmith.init. Call flagsmith.init first, if you wish to prevent it sending a request for flags, call init with preventFetch:true**.```


any idea ?


```
import flagsmith from "flagsmith"; 
flagsmith.identify("user_123457"); // This will create a user in the dashboard if they don't already exist

flagsmith.setTrait("age", 21);
flagsmith.incrementTrait("age", -1);
flagsmith.init({
  environmentID:"mTsSYnjxfkDUYKTYHqMdwx",
  api:"http://localhost:8000/api/v1/",
  onChange: (oldFlags, params) => { // Occurs whenever flags are changed

      const { isFromServer } = params; // Determines if the update came from the server or local cached storage

      // Check for a feature
      if (flagsmith.hasFeature("intl_shipping")) {
          console.log("got feature")
      }

      // Or, use the value of a feature
      const intl_shipping = flagsmith.getValue("intl_shipping");
      console.log('intl_shipping =>',intl_shipping )
      const age = flagsmith.getTrait("age");
      console.log('age trait',age);
      // Check whether value has changed
      const intl_shipping_Old = oldFlags["intl_shipping"] && oldFlags["intl_shipping"].value;
      console.log('intl_shipping_Old =>',intl_shipping_Old )
      if (intl_shipping !== intl_shipping_Old) {
          // Value has changed, do something here
          console.log('intl_shipping =>',intl_shipping , 'value changed')
      }
  }
});
```

**Reply by dabeeeenster** [reply]
You need to `flagsmith.init` before `flagsmith.setTrait` or `flagsmith.incrementTrait `.

**Reply by khirod-dev** [reply]
Thank you . @dabeeeenster 
So when so see the docs for creating traits the sample code shows the way I have here . So, do the docs need an update ? 
And my 2nd question :- anywhere during the application flow we should be able to set the trait on some use behavior, does  that mean before I setTrait always  call init and then set.  even if I am doing at 3,4 different places during app flow ?

**Comment by dabeeeenster** [comment]
Ok thanks for the feedback - this example is updated
